document/chapters/query_language.tex

```diff
@@ -187,14 +187,48 @@ \subsection{Comments and whitespace}
 \section{Scoping and Typing}
 \label{sec:scoping_and_typing}
 
-\mnote{TODO: Ann Roy}
+\subsection{Variables}
+The \vnnlib{} query language, utilizes variables to refer to specific inputs, outputs, and hidden outputs for neural networks.
+
+\paragraph{Declaration}
+Variable declarations bind a variable name to a specific ONNX tensor, which is then used in the \textit{Assertions} section of the query. Variables are declared into a query context in the \textit{Network Declarations} section of a \vnnlib{} query with a \textit{name}, and other ONNX \textit{Tensor} specfic properties such as: ONNX element type, shape, and optionally, the ONNX node name. Variables are also declared in order of input, hidden, and output declarations within a network declaration.
+
+\paragraph{Referencing}
+Variables are referenced within the \textit{Assertions} section of a query using its name, accompanied by the tensor indices to refer to an element in the corresponding tensor.
+
+
+
+\myremark {
+    Variable names are also used in \textit{Network Declarations} to refer to specific networks, but these are not used within the body of the query, and instead used for the purpose of externally binding the network to a specific ONNX model file during verification.
+    
+    Similarly, the ONNX node names are used to refer to specific nodes within the ONNX model file, but these are not used within the body of the query.
+}
```

Comment:
All of the stuff above here should be merged into the syntax text if the information isn't there already.